File: templates/python/bar_lollipop.py
"""bar_lollipop: 棒棒糖图（替代柱状，更轻盈）."""
import sys, os; sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', '_utils', 'python'))
import numpy as np
import matplotlib.pyplot as plt
from theme import apply_theme
from palette import cycle

def make_figure(labels=None, values=None, title='Lollipop'):
    apply_theme()
    if labels is None:
        labels = [f'item {i+1}' for i in range(10)]
        values = np.random.default_rng(3).uniform(20, 90, 10)
    order = np.argsort(values)
    labels = [labels[i] for i in order]; values = values[order]
    y = np.arange(len(labels))
    fig, ax = plt.subplots()
    ax.hlines(y, 0, values, color=cycle(0), linewidth=1.5)
    ax.plot(values, y, 'o', color=cycle(0), markersize=8)
    ax.set_yticks(y); ax.set_yticklabels(labels)
    ax.set_xlabel('value'); ax.set_title(title)
    ax.grid(True, axis='x', linestyle=':', alpha=0.5)
    fig.tight_layout()
    return fig

if __name__ == '__main__':
    fig = make_figure()
    fig.savefig(__file__.replace('.py', '.png'), dpi=150)

File: templates/matlab/bar_lollipop.m
function fig = bar_lollipop()
    addpath(fullfile(fileparts(mfilename('fullpath')),'..','..','_utils','matlab'));
    apply_theme(); rng(3); c = palette('cat',1);
    labels = arrayfun(@(i)sprintf('item %d',i),1:10,'UniformOutput',false);
    v = 20 + 70*rand(1, 10); [v, idx] = sort(v); labels = labels(idx);
    y = 1:numel(v);
    fig = figure; hold on;
    for i = 1:numel(v)
        plot([0 v(i)], [y(i) y(i)], 'Color', c, 'LineWidth', 1.5);
    end
    plot(v, y, 'o', 'Color', c, 'MarkerFaceColor', c, 'MarkerSize', 8);
    set(gca,'YTick',y,'YTickLabel',labels);
    xlabel('value'); title('Lollipop'); grid on;
end
